mautic/mautic
# Redis sentinel_master_only option

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [❌]
| New feature/enhancement? (use the a.x branch)      | [✅]
| Deprecations?                          | [❌]
| BC breaks? (use the c.x branch)        | [❌]
| Automated tests included?              | [✅] 
| Related user documentation PR URL      |  
| Related developer documentation PR URL |  
| Issue(s) addressed                     |  



#### Description:
This PR introduces the new configuration option `'redis_primary_only'` for Redis sentinel setup. If the option is enabled then the Redis client performs **read operations** (by default read operations reach slave nodes while write operations reach a master node) against a master Redis node.

#### Steps to test this PR:

1. Testing this PR requires some Redis expertise.
2. Configure a mautic instance with a single Redis node and make sure there are no issues introduced.
3. Configure a mautic instance with a Sentinel Redis node using the `'redis_primary_only' => true` and make sure that all the requests go to the master node.
